diff --git a/devel/py-gitpython/Makefile b/devel/py-gitpython/Makefile
index 12614ef9b945..34c38e12976b 100644
--- a/devel/py-gitpython/Makefile
+++ b/devel/py-gitpython/Makefile
@@ -1,6 +1,6 @@
 # $FreeBSD$
 
-PORTNAME=	GitPython
+PORTNAME=	gitpython
 DISTVERSION=	3.1.11
 CATEGORIES=	devel python
 MASTER_SITES=	CHEESESHOP
diff --git a/pkglist b/pkglist
new file mode 100644
index 000000000000..6339616e0217
--- /dev/null
+++ b/pkglist
@@ -0,0 +1 @@
+devel/py-gitpython
