diff --git a/devel/R-cran-brio/Makefile b/devel/R-cran-brio/Makefile
index b51a9c0ee2ce..f06cd0096ef9 100644
--- a/devel/R-cran-brio/Makefile
+++ b/devel/R-cran-brio/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
 PORTNAME=	brio
-DISTVERSION=	1.1.0
+DISTVERSION=	1.1.1
 CATEGORIES=	devel
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
diff --git a/devel/R-cran-brio/distinfo b/devel/R-cran-brio/distinfo
index ddcd255cb34a..7d851fcdf698 100644
--- a/devel/R-cran-brio/distinfo
+++ b/devel/R-cran-brio/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1608300467
-SHA256 (brio_1.1.0.tar.gz) = 6bb3a3b47bea13f1a1e3dcdc8b9f688502643e4b40a481a34aa04a261aabea38
-SIZE (brio_1.1.0.tar.gz) = 11610
+TIMESTAMP = 1611457172
+SHA256 (brio_1.1.1.tar.gz) = 70fa42423fc2ef8ec5ef91ca2d6ed3a1c79200d650774133c821cc5f38be222a
+SIZE (brio_1.1.1.tar.gz) = 11652
