diff --git a/devel/py-codecov/Makefile b/devel/py-codecov/Makefile
index 720955eb7c41..bb6dcbfc1b4d 100644
--- a/devel/py-codecov/Makefile
+++ b/devel/py-codecov/Makefile
@@ -2,7 +2,7 @@
 # $FreeBSD$
 
 PORTNAME=	codecov
-PORTVERSION=	2.1.10
+PORTVERSION=	2.1.11
 DISTVERSIONPREFIX=	v
 CATEGORIES=	devel python
 PKGNAMEPREFIX=	${PYTHON_PKGNAMEPREFIX}
diff --git a/devel/py-codecov/distinfo b/devel/py-codecov/distinfo
index 6a2083db29b7..f728ab6c8b24 100644
--- a/devel/py-codecov/distinfo
+++ b/devel/py-codecov/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1606306150
-SHA256 (codecov-codecov-python-v2.1.10_GH0.tar.gz) = 00a4d3272612363dd34eec0fa1f7d9d1ae1e95c9c7c92c60b8783400855fe16d
-SIZE (codecov-codecov-python-v2.1.10_GH0.tar.gz) = 26275
+TIMESTAMP = 1608741539
+SHA256 (codecov-codecov-python-v2.1.11_GH0.tar.gz) = 8c749dad020287910c40a8900370a2e88ff1c0b2519eaf28054511008425e186
+SIZE (codecov-codecov-python-v2.1.11_GH0.tar.gz) = 26424
