diff --git a/devel/Makefile b/devel/Makefile
index a0ad371b87ba..f02bbf5195f7 100644
--- a/devel/Makefile
+++ b/devel/Makefile
@@ -51,6 +51,7 @@
     SUBDIR += R-cran-gsubfn
     SUBDIR += R-cran-gtools
     SUBDIR += R-cran-hms
+    SUBDIR += R-cran-ini
     SUBDIR += R-cran-iterators
     SUBDIR += R-cran-itertools
     SUBDIR += R-cran-later
diff --git a/devel/R-cran-ini/Makefile b/devel/R-cran-ini/Makefile
new file mode 100644
index 000000000000..68cf71d8ac9e
--- /dev/null
+++ b/devel/R-cran-ini/Makefile
@@ -0,0 +1,17 @@
+# $FreeBSD$
+
+PORTNAME=	ini
+DISTVERSION=	0.3.1
+CATEGORIES=	devel
+DISTNAME=	${PORTNAME}_${DISTVERSION}
+
+MAINTAINER=	ygy@FreeBSD.org
+COMMENT=	Read and Write '.ini' Files
+
+LICENSE=	GPLv3
+
+TEST_DEPENDS=	R-cran-testthat>0:devel/R-cran-testthat
+
+USES=		cran:auto-plist
+
+.include <bsd.port.mk>
diff --git a/devel/R-cran-ini/distinfo b/devel/R-cran-ini/distinfo
new file mode 100644
index 000000000000..966845cacde0
--- /dev/null
+++ b/devel/R-cran-ini/distinfo
@@ -0,0 +1,3 @@
+TIMESTAMP = 1609012627
+SHA256 (ini_0.3.1.tar.gz) = 7b191a54019c8c52d6c2211c14878c95564154ec4865f57007953742868cd813
+SIZE (ini_0.3.1.tar.gz) = 3488
diff --git a/devel/R-cran-ini/pkg-descr b/devel/R-cran-ini/pkg-descr
new file mode 100644
index 000000000000..72784fc37af3
--- /dev/null
+++ b/devel/R-cran-ini/pkg-descr
@@ -0,0 +1,5 @@
+Parse simple '.ini' configuration files to an structured list. Users can
+manipulate this resulting list with lapply() functions. This same structured
+list can be used to write back to file after modifications.
+
+WWW: https://github.com/dvdscripter/ini
