diff --git a/devel/R-cran-covr/Makefile b/devel/R-cran-covr/Makefile
index 605b0eaefd2f..d7814c048670 100644
--- a/devel/R-cran-covr/Makefile
+++ b/devel/R-cran-covr/Makefile
@@ -1,12 +1,11 @@
 # $FreeBSD$
 
 PORTNAME=	covr
-DISTVERSION=	3.2.1
-PORTREVISION=	3
+DISTVERSION=	3.5.1
 CATEGORIES=	devel
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
-MAINTAINER=	ports@FreeBSD.org
+MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Test Coverage for Packages
 
 LICENSE=	GPLv3
diff --git a/devel/R-cran-covr/distinfo b/devel/R-cran-covr/distinfo
index aa1ed2091869..43277472b3d9 100644
--- a/devel/R-cran-covr/distinfo
+++ b/devel/R-cran-covr/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1540318008
-SHA256 (covr_3.2.1.tar.gz) = ea90daa48011e4ac4431ae47ee02fad98f54b529fc3900281cbeef7a2edef0a0
-SIZE (covr_3.2.1.tar.gz) = 140046
+TIMESTAMP = 1606834269
+SHA256 (covr_3.5.1.tar.gz) = a54cfc3623ea56084158ac5d7fe33f216f45191f6dcddab9c9ed4ec1d9d8ac6c
+SIZE (covr_3.5.1.tar.gz) = 146686
