diff --git a/devel/py-gitpython/Makefile b/devel/py-gitpython/Makefile
index 117fc3b6bc1c..7cb3e5945cf5 100644
--- a/devel/py-gitpython/Makefile
+++ b/devel/py-gitpython/Makefile
@@ -1,8 +1,9 @@
-PORTNAME=	GitPython
-DISTVERSION=	3.1.11
+PORTNAME=	gitpython
+DISTVERSION=	3.1.15
 CATEGORIES=	devel python
 MASTER_SITES=	CHEESESHOP
 PKGNAMEPREFIX=	${PYTHON_PKGNAMEPREFIX}
+DISTNAME=	GitPython-${PORTVERSION}
 
 MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Python Git Library
diff --git a/devel/py-gitpython/distinfo b/devel/py-gitpython/distinfo
index 5ed68c7c9e03..76cf83576ab4 100644
--- a/devel/py-gitpython/distinfo
+++ b/devel/py-gitpython/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1605172201
-SHA256 (GitPython-3.1.11.tar.gz) = befa4d101f91bad1b632df4308ec64555db684c360bd7d2130b4807d49ce86b8
-SIZE (GitPython-3.1.11.tar.gz) = 170908
+TIMESTAMP = 1619622145
+SHA256 (GitPython-3.1.15.tar.gz) = 05af150f47a5cca3f4b0af289b73aef8cf3c4fe2385015b06220cbcdee48bb6e
+SIZE (GitPython-3.1.15.tar.gz) = 175132
