diff --git a/devel/py-gitless/Makefile b/devel/py-gitless/Makefile
index 6d6518667739..b94b4c1a3eef 100644
--- a/devel/py-gitless/Makefile
+++ b/devel/py-gitless/Makefile
@@ -2,13 +2,12 @@
 # $FreeBSD$
 
 PORTNAME=	gitless
-PORTVERSION=	0.8.6
+PORTVERSION=	0.8.8
 DISTVERSIONPREFIX=	v
-PORTREVISION=	3
 CATEGORIES=	devel python
 PKGNAMEPREFIX=	${PYTHON_PKGNAMEPREFIX}
 
-MAINTAINER=	ports@FreeBSD.org
+MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Version control system built on top of Git
 
 LICENSE=	MIT
diff --git a/devel/py-gitless/distinfo b/devel/py-gitless/distinfo
index 0a1b87f40ea0..f453f33d5e8e 100644
--- a/devel/py-gitless/distinfo
+++ b/devel/py-gitless/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1518205276
-SHA256 (sdg-mit-gitless-v0.8.6_GH0.tar.gz) = e1d009bf9d7c89428d7029394cc85a0d91bd2af73f019508ddc92c98faeed8e5
-SIZE (sdg-mit-gitless-v0.8.6_GH0.tar.gz) = 41764
+TIMESTAMP = 1606304709
+SHA256 (sdg-mit-gitless-v0.8.8_GH0.tar.gz) = 470aab13d51baec2ab54d7ceb6d12b9a2937f72d840516affa0cb34a6360523c
+SIZE (sdg-mit-gitless-v0.8.8_GH0.tar.gz) = 42432
