diff --git a/devel/ccls/Makefile b/devel/ccls/Makefile
index 33e273493414..9816fe2b1399 100644
--- a/devel/ccls/Makefile
+++ b/devel/ccls/Makefile
@@ -1,19 +1,17 @@
 # $FreeBSD$
 
 PORTNAME=	ccls
-DISTVERSION=	0.20190823.4
+DISTVERSION=	0.20201219
 CATEGORIES=	devel
-MASTER_SITES=	LOCAL/tobik
 
-MAINTAINER=	ports@FreeBSD.org
+MAINTAINER=	khng300@gmail.com
 COMMENT=	C/C++/ObjC language server
 
 LICENSE=	APACHE20
 LICENSE_FILE=	${WRKSRC}/LICENSE
 
-BUILD_DEPENDS=	llvm${LLVM_DEFAULT}>0:devel/llvm${LLVM_DEFAULT} \
-		rapidjson>=1.1.0_4:devel/rapidjson
-LIB_DEPENDS=	libLLVM.so:devel/llvm${LLVM_DEFAULT}
+BUILD_DEPENDS=	rapidjson>=1.1.0_4:devel/rapidjson
+LIB_DEPENDS=	libclang-cpp.so:devel/llvm${LLVM_DEFAULT}
 
 USES=		cmake compiler:c++17-lang localbase
 USE_GITHUB=	yes
diff --git a/devel/ccls/distinfo b/devel/ccls/distinfo
index 76ed9bed11ee..912b4737979a 100644
--- a/devel/ccls/distinfo
+++ b/devel/ccls/distinfo
@@ -1,5 +1,5 @@
-TIMESTAMP = 1572679443
+TIMESTAMP = 1609391939
 SHA256 (MaskRay-ccls.wiki-20190903.tar.gz) = da8d2ed4464ca445001eb2b22439ab265314ad8cf86b4b9509ac8a6a19c481e8
 SIZE (MaskRay-ccls.wiki-20190903.tar.gz) = 32080
-SHA256 (MaskRay-ccls-0.20190823.4_GH0.tar.gz) = 19f394b69d70415d3be9a11a030e0bb28ddc2b654cbfa35f4570096935f9f536
-SIZE (MaskRay-ccls-0.20190823.4_GH0.tar.gz) = 158811
+SHA256 (MaskRay-ccls-0.20201219_GH0.tar.gz) = edd3435bc7e55d9e5dc931932f9c98275a6a28d1ab1f66416110e029f3f2882a
+SIZE (MaskRay-ccls-0.20201219_GH0.tar.gz) = 160445
diff --git a/devel/ccls/files/patch-src_indexer.cc b/devel/ccls/files/patch-src_indexer.cc
index afac210fe60c..e69de29bb2d1 100644
--- a/devel/ccls/files/patch-src_indexer.cc
+++ b/devel/ccls/files/patch-src_indexer.cc
@@ -1,22 +0,0 @@
-https://github.com/MaskRay/ccls/commit/b0d42ad8f2ef22d3a42eee6e7534fc363c018440
-
---- src/indexer.cc.orig	2019-10-25 01:37:55 UTC
-+++ src/indexer.cc
-@@ -677,10 +677,13 @@ class IndexDataConsumer : public index::IndexDataConsu
- public:
-   IndexDataConsumer(IndexParam &param) : param(param) {}
-   void initialize(ASTContext &ctx) override { this->ctx = param.ctx = &ctx; }
--  bool handleDeclOccurence(const Decl *d, index::SymbolRoleSet roles,
--                           ArrayRef<index::SymbolRelation> relations,
--                           SourceLocation src_loc,
--                           ASTNodeInfo ast_node) override {
-+#if LLVM_VERSION_MAJOR < 10 // llvmorg-10-init-12036-g3b9715cb219
-+# define handleDeclOccurrence handleDeclOccurence
-+#endif
-+  bool handleDeclOccurrence(const Decl *d, index::SymbolRoleSet roles,
-+                            ArrayRef<index::SymbolRelation> relations,
-+                            SourceLocation src_loc,
-+                            ASTNodeInfo ast_node) override {
-     if (!param.no_linkage) {
-       if (auto *nd = dyn_cast<NamedDecl>(d); nd && nd->hasLinkage())
-         ;
