diff --git a/devel/Makefile b/devel/Makefile
index 0c30dd0b651c..139123e19a54 100644
--- a/devel/Makefile
+++ b/devel/Makefile
@@ -41,6 +41,7 @@
     SUBDIR += R-cran-ellipsis
     SUBDIR += R-cran-evaluate
     SUBDIR += R-cran-fansi
+    SUBDIR += R-cran-filelock
     SUBDIR += R-cran-foreach
     SUBDIR += R-cran-gbRd
     SUBDIR += R-cran-gbm
diff --git a/devel/R-cran-filelock/Makefile b/devel/R-cran-filelock/Makefile
new file mode 100644
index 000000000000..ee80a31e23ec
--- /dev/null
+++ b/devel/R-cran-filelock/Makefile
@@ -0,0 +1,21 @@
+# Created by: Guangyuan Yang <ygy@FreeBSD.org>
+# $FreeBSD$
+
+PORTNAME=	filelock
+DISTVERSION=	1.0.2
+CATEGORIES=	devel
+DISTNAME=	${PORTNAME}_${DISTVERSION}
+
+MAINTAINER=	ygy@FreeBSD.org
+COMMENT=	Portable File Locking
+
+LICENSE=	MIT
+LICENSE_FILE=	${WRKSRC}/LICENSE
+
+TEST_DEPENDS=	R-cran-callr>=2.0.0:devel/R-cran-callr \
+		R-cran-covr>0:devel/R-cran-covr \
+		R-cran-testthat>0:devel/R-cran-testthat
+
+USES=		cran:auto-plist,compiles
+
+.include <bsd.port.mk>
diff --git a/devel/R-cran-filelock/distinfo b/devel/R-cran-filelock/distinfo
new file mode 100644
index 000000000000..a590ef6b87f4
--- /dev/null
+++ b/devel/R-cran-filelock/distinfo
@@ -0,0 +1,3 @@
+TIMESTAMP = 1610211187
+SHA256 (filelock_1.0.2.tar.gz) = ac2915950789b16c43a625a2b8dab6ba423588db4a7d0daa75b74518b82b1403
+SIZE (filelock_1.0.2.tar.gz) = 12377
diff --git a/devel/R-cran-filelock/pkg-descr b/devel/R-cran-filelock/pkg-descr
new file mode 100644
index 000000000000..6134694f56d5
--- /dev/null
+++ b/devel/R-cran-filelock/pkg-descr
@@ -0,0 +1,4 @@
+Place an exclusive or shared lock on a file. It uses 'LockFile' on Windows and
+'fcntl' locks on Unix-like systems.
+
+WWW: https://github.com/r-lib/filelock#readme
