diff --git a/devel/py-gitpython/Makefile b/devel/py-gitpython/Makefile
index 94684b9c22dc..408c4a4601f6 100644
--- a/devel/py-gitpython/Makefile
+++ b/devel/py-gitpython/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
 PORTNAME=	GitPython
-DISTVERSION=	2.1.11
+DISTVERSION=	3.1.11
 CATEGORIES=	devel python
 MASTER_SITES=	CHEESESHOP
 PKGNAMEPREFIX=	${PYTHON_PKGNAMEPREFIX}
diff --git a/devel/py-gitpython/distinfo b/devel/py-gitpython/distinfo
index 66101f718d68..5ed68c7c9e03 100644
--- a/devel/py-gitpython/distinfo
+++ b/devel/py-gitpython/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1545318709
-SHA256 (GitPython-2.1.11.tar.gz) = 8237dc5bfd6f1366abeee5624111b9d6879393d84745a507de0fda86043b65a8
-SIZE (GitPython-2.1.11.tar.gz) = 428531
+TIMESTAMP = 1605172201
+SHA256 (GitPython-3.1.11.tar.gz) = befa4d101f91bad1b632df4308ec64555db684c360bd7d2130b4807d49ce86b8
+SIZE (GitPython-3.1.11.tar.gz) = 170908
