diff --git a/devel/R-cran-brio/Makefile b/devel/R-cran-brio/Makefile
index 2dcc029fab58..3bb0eb0b88be 100644
--- a/devel/R-cran-brio/Makefile
+++ b/devel/R-cran-brio/Makefile
@@ -1,5 +1,5 @@
 PORTNAME=	brio
-DISTVERSION=	1.1.1
+DISTVERSION=	1.1.2
 CATEGORIES=	devel
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
diff --git a/devel/R-cran-brio/distinfo b/devel/R-cran-brio/distinfo
index 7d851fcdf698..7bbe0537625d 100644
--- a/devel/R-cran-brio/distinfo
+++ b/devel/R-cran-brio/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1611457172
-SHA256 (brio_1.1.1.tar.gz) = 70fa42423fc2ef8ec5ef91ca2d6ed3a1c79200d650774133c821cc5f38be222a
-SIZE (brio_1.1.1.tar.gz) = 11652
+TIMESTAMP = 1619568183
+SHA256 (brio_1.1.2.tar.gz) = 42dde6953151e31cc38bbec72335c01ac9e755cc07d11e26f4e1fcd0f9f471ef
+SIZE (brio_1.1.2.tar.gz) = 12069
