diff --git a/devel/gradle/Makefile b/devel/gradle/Makefile
index a5d076712e23..db732543e66d 100644
--- a/devel/gradle/Makefile
+++ b/devel/gradle/Makefile
@@ -2,7 +2,7 @@
 # $FreeBSD$
 
 PORTNAME=	gradle
-PORTVERSION=	6.7.1
+PORTVERSION=	6.8
 CATEGORIES=	devel java
 MASTER_SITES=	https://downloads.gradle.org/distributions/
 EXTRACT_SUFX=	-bin.zip
diff --git a/devel/gradle/distinfo b/devel/gradle/distinfo
index 6a3db9c7ccbb..21050e1037b9 100644
--- a/devel/gradle/distinfo
+++ b/devel/gradle/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1607931114
-SHA256 (gradle-6.7.1-bin.zip) = 3239b5ed86c3838a37d983ac100573f64c1f3fd8e1eb6c89fa5f9529b5ec091d
-SIZE (gradle-6.7.1-bin.zip) = 102842885
+TIMESTAMP = 1610938239
+SHA256 (gradle-6.8-bin.zip) = e2774e6fb77c43657decde25542dea710aafd78c4022d19b196e7e78d79d8c6c
+SIZE (gradle-6.8-bin.zip) = 107841145
diff --git a/devel/gradle/pkg-descr b/devel/gradle/pkg-descr
index 32f6de448844..9ca5da488bc7 100644
--- a/devel/gradle/pkg-descr
+++ b/devel/gradle/pkg-descr
@@ -13,4 +13,4 @@ sensible defaults. Gradle is quickly becoming the build system of
 choice for many open source projects, leading edge enterprises and
 legacy automation challenges.
 
-WWW: http://www.gradle.org/
+WWW: https://www.gradle.org/
