diff --git a/lang/kotlin/Makefile b/lang/kotlin/Makefile
index 989dade9d54d..643af754e08a 100644
--- a/lang/kotlin/Makefile
+++ b/lang/kotlin/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
 PORTNAME=	kotlin
-PORTVERSION=	1.4.10
+PORTVERSION=	1.4.20
 CATEGORIES=	lang
 MASTER_SITES=	https://github.com/JetBrains/kotlin/releases/download/v${PORTVERSION}/
 DISTNAME=	kotlin-compiler-${PORTVERSION}
diff --git a/lang/kotlin/distinfo b/lang/kotlin/distinfo
index c23de797ded2..7f04e78bf0e4 100644
--- a/lang/kotlin/distinfo
+++ b/lang/kotlin/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1600654868
-SHA256 (kotlin-compiler-1.4.10.zip) = bb1a21d70e521a01ae104e99a082a6e7bb58699b86347049da521d175d0dace7
-SIZE (kotlin-compiler-1.4.10.zip) = 54526905
+TIMESTAMP = 1608361542
+SHA256 (kotlin-compiler-1.4.20.zip) = 11db93a4d6789e3406c7f60b9f267eba26d6483dcd771eff9f85bb7e9837011f
+SIZE (kotlin-compiler-1.4.20.zip) = 60773121
diff --git a/lang/kotlin/pkg-plist b/lang/kotlin/pkg-plist
index 213dce5005ca..044148c447b6 100644
--- a/lang/kotlin/pkg-plist
+++ b/lang/kotlin/pkg-plist
@@ -64,5 +64,7 @@ bin/kotlinc-jvm
 %%DATADIR%%/lib/kotlinx-serialization-compiler-plugin.jar
 %%DATADIR%%/lib/mutability-annotations-compat.jar
 %%DATADIR%%/lib/noarg-compiler-plugin.jar
+%%DATADIR%%/lib/parcelize-compiler.jar
+%%DATADIR%%/lib/parcelize-runtime.jar
 %%DATADIR%%/lib/sam-with-receiver-compiler-plugin.jar
 %%DATADIR%%/lib/trove4j.jar
