diff --git a/net/py-eventlet/Makefile b/net/py-eventlet/Makefile
index f205169afc55..e88ede561289 100644
--- a/net/py-eventlet/Makefile
+++ b/net/py-eventlet/Makefile
@@ -2,12 +2,12 @@
 # $FreeBSD$
 
 PORTNAME=	eventlet
-PORTVERSION=	0.28.0
+PORTVERSION=	0.29.1
 CATEGORIES=	net python
 MASTER_SITES=	CHEESESHOP
 PKGNAMEPREFIX=	${PYTHON_PKGNAMEPREFIX}
 
-MAINTAINER=	ports@FreeBSD.org
+MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Concurrent networking library for Python
 
 LICENSE=	MIT
diff --git a/net/py-eventlet/distinfo b/net/py-eventlet/distinfo
index 94e3751784f5..751fe5d7c86d 100644
--- a/net/py-eventlet/distinfo
+++ b/net/py-eventlet/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1601764290
-SHA256 (eventlet-0.28.0.tar.gz) = 9c7c63e8a80c7d02d692b2cf308312402fa4777335fd3de5da45097383301ff3
-SIZE (eventlet-0.28.0.tar.gz) = 399360
+TIMESTAMP = 1607485128
+SHA256 (eventlet-0.29.1.tar.gz) = 9faff63631b01277c463ae91cd4ab3f25a2f0f5abe3219d43a386ef1daa6159a
+SIZE (eventlet-0.29.1.tar.gz) = 400673
