diff --git a/security/step-cli/Makefile b/security/step-cli/Makefile
index 0a63da9b0f11..71e6b2ab620f 100644
--- a/security/step-cli/Makefile
+++ b/security/step-cli/Makefile
@@ -2,7 +2,7 @@
 
 PORTNAME=	step-cli
 DISTVERSIONPREFIX=v
-DISTVERSION=	0.15.2
+DISTVERSION=	0.15.3
 CATEGORIES=	security
 
 MAINTAINER=	mw@wipp.bayern
@@ -31,6 +31,7 @@ GH_TUPLE=	\
 		bgentry:speakeasy:v0.1.0:bgentry_speakeasy/vendor/github.com/bgentry/speakeasy \
 		bombsimon:wsl:v2.0.0:bombsimon_wsl_v2/vendor/github.com/bombsimon/wsl/v2 \
 		boombuler:barcode:v1.0.0:boombuler_barcode/vendor/github.com/boombuler/barcode \
+		census-instrumentation:opencensus-go:v0.22.4:census_instrumentation_opencensus_go/vendor/go.opencensus.io \
 		cespare:xxhash:v1.1.0:cespare_xxhash/vendor/github.com/cespare/xxhash \
 		cespare:xxhash:v2.1.1:cespare_xxhash_v2/vendor/github.com/cespare/xxhash/v2 \
 		cheggaaa:pb:v1.0.28:cheggaaa_pb/vendor/gopkg.in/cheggaaa/pb.v1 \
@@ -49,7 +50,7 @@ GH_TUPLE=	\
 		dgraph-io:ristretto:8f368f2f2ab3:dgraph_io_ristretto/vendor/github.com/dgraph-io/ristretto \
 		dgrijalva:jwt-go:v3.2.0:dgrijalva_jwt_go/vendor/github.com/dgrijalva/jwt-go \
 		dgryski:go-farm:6a90982ecee2:dgryski_go_farm/vendor/github.com/dgryski/go-farm \
-		dominikh:go-tools:v0.0.1-2019.2.3:dominikh_go_tools/vendor/honnef.co/go/tools \
+		dominikh:go-tools:v0.0.1-2020.1.4:dominikh_go_tools/vendor/honnef.co/go/tools \
 		dustin:go-humanize:v1.0.0:dustin_go_humanize/vendor/github.com/dustin/go-humanize \
 		etcd-io:bbolt:v1.3.2:etcd_io_bbolt/vendor/go.etcd.io/bbolt \
 		etcd-io:etcd:v3.3.18:etcd_io_etcd \
@@ -71,18 +72,22 @@ GH_TUPLE=	\
 		gobwas:glob:v0.2.3:gobwas_glob/vendor/github.com/gobwas/glob \
 		gofrs:flock:5135e617513b:gofrs_flock/vendor/github.com/gofrs/flock \
 		gogo:protobuf:v1.3.1:gogo_protobuf/vendor/github.com/gogo/protobuf \
+		golang:appengine:v1.6.6:golang_appengine/vendor/google.golang.org/appengine \
 		golang:crypto:123391ffb6de:golang_crypto/vendor/golang.org/x/crypto \
 		golang:glog:23def4e6c14b:golang_glog/vendor/github.com/golang/glog \
-		golang:groupcache:215e87163ea7:golang_groupcache/vendor/github.com/golang/groupcache \
-		golang:lint:fdd1cda4f05f:golang_lint/vendor/golang.org/x/lint \
-		golang:mock:v1.3.1:golang_mock/vendor/github.com/golang/mock \
-		golang:net:16171245cfb2:golang_net/vendor/golang.org/x/net \
-		golang:protobuf:v1.3.2:golang_protobuf/vendor/github.com/golang/protobuf \
+		golang:groupcache:8c9f03a8e57e:golang_groupcache/vendor/github.com/golang/groupcache \
+		golang:lint:738671d3881b:golang_lint/vendor/golang.org/x/lint \
+		golang:mock:v1.4.4:golang_mock/vendor/github.com/golang/mock \
+		golang:mod:v0.3.0:golang_mod/vendor/golang.org/x/mod \
+		golang:net:c89045814202:golang_net/vendor/golang.org/x/net \
+		golang:oauth2:bf48bf16ab8d:golang_oauth2/vendor/golang.org/x/oauth2 \
+		golang:protobuf:v1.4.2:golang_protobuf/vendor/github.com/golang/protobuf \
 		golang:snappy:v0.0.1:golang_snappy/vendor/github.com/golang/snappy \
-		golang:sys:b016eb3dc98e:golang_sys/vendor/golang.org/x/sys \
-		golang:text:v0.3.2:golang_text/vendor/golang.org/x/text \
+		golang:sys:157a740278f4:golang_sys/vendor/golang.org/x/sys \
+		golang:text:v0.3.3:golang_text/vendor/golang.org/x/text \
 		golang:time:555d28b269f0:golang_time/vendor/golang.org/x/time \
-		golang:tools:7be0a674c9fc:golang_tools/vendor/golang.org/x/tools \
+		golang:tools:af4cc2cd812e:golang_tools/vendor/golang.org/x/tools \
+		golang:xerrors:5ec99f83aff1:golang_xerrors/vendor/golang.org/x/xerrors \
 		golangci:check:cfe4005ccda2:golangci_check/vendor/github.com/golangci/check \
 		golangci:dupl:3e9179ac440a:golangci_dupl/vendor/github.com/golangci/dupl \
 		golangci:errcheck:ef45e06d44b6:golangci_errcheck/vendor/github.com/golangci/errcheck \
@@ -100,15 +105,19 @@ GH_TUPLE=	\
 		golangci:unconvert:28b1c447d1f4:golangci_unconvert/vendor/github.com/golangci/unconvert \
 		google:btree:v1.0.0:google_btree/vendor/github.com/google/btree \
 		google:certificate-transparency-go:v1.1.0:google_certificate_transparency_go/vendor/github.com/google/certificate-transparency-go \
-		google:go-genproto:f3c370f40bfb:google_go_genproto/vendor/google.golang.org/genproto \
+		google:go-cmp:v0.5.2:google_go_cmp/vendor/github.com/google/go-cmp \
+		google:go-genproto:0bd0a958aa1d:google_go_genproto/vendor/google.golang.org/genproto \
 		google:monologue:35abc9683a6c:google_monologue/vendor/github.com/google/monologue \
-		google:uuid:v1.1.1:google_uuid/vendor/github.com/google/uuid \
+		google:uuid:v1.1.2:google_uuid/vendor/github.com/google/uuid \
+		googleapis:gax-go:v2.0.5:googleapis_gax_go_v2/vendor/github.com/googleapis/gax-go \
+		googleapis:google-api-go-client:v0.31.0:googleapis_google_api_go_client/vendor/google.golang.org/api \
+		googleapis:google-cloud-go:3c2db50b5678:googleapis_google_cloud_go/vendor/cloud.google.com/go \
 		gorilla:websocket:v1.4.1:gorilla_websocket/vendor/github.com/gorilla/websocket \
 		gostaticanalysis:analysisutil:v0.0.3:gostaticanalysis_analysisutil/vendor/github.com/gostaticanalysis/analysisutil \
-		grpc-ecosystem:go-grpc-middleware:v1.1.0:grpc_ecosystem_go_grpc_middleware/vendor/github.com/grpc-ecosystem/go-grpc-middleware \
+			grpc-ecosystem:go-grpc-middleware:v1.1.0:grpc_ecosystem_go_grpc_middleware/vendor/github.com/grpc-ecosystem/go-grpc-middleware \
 		grpc-ecosystem:go-grpc-prometheus:v1.2.0:grpc_ecosystem_go_grpc_prometheus/vendor/github.com/grpc-ecosystem/go-grpc-prometheus \
 		grpc-ecosystem:grpc-gateway:v1.12.1:grpc_ecosystem_grpc_gateway/vendor/github.com/grpc-ecosystem/grpc-gateway \
-		grpc:grpc-go:v1.26.0:grpc_grpc_go/vendor/google.golang.org/grpc \
+		grpc:grpc-go:v1.29.1:grpc_grpc_go/vendor/google.golang.org/grpc \
 		hashicorp:hcl:v1.0.0:hashicorp_hcl/vendor/github.com/hashicorp/hcl \
 		huandu:xstrings:v1.3.1:huandu_xstrings/vendor/github.com/huandu/xstrings \
 		icrowley:fake:4178557ae428:icrowley_fake/vendor/github.com/icrowley/fake \
@@ -150,6 +159,7 @@ GH_TUPLE=	\
 		prometheus:client_model:v0.1.0:prometheus_client_model/vendor/github.com/prometheus/client_model \
 		prometheus:common:v0.7.0:prometheus_common/vendor/github.com/prometheus/common \
 		prometheus:procfs:v0.0.8:prometheus_procfs/vendor/github.com/prometheus/procfs \
+		protocolbuffers:protobuf-go:v1.25.0:protocolbuffers_protobuf_go/vendor/google.golang.org/protobuf \
 		rs:xid:v1.2.1:rs_xid/vendor/github.com/rs/xid \
 		russross:blackfriday:v2.0.1:russross_blackfriday_v2/vendor/github.com/russross/blackfriday/v2 \
 		samfoo:ansi:b6bd2ded7189:samfoo_ansi/vendor/github.com/samfoo/ansi \
@@ -157,9 +167,9 @@ GH_TUPLE=	\
 		shurcooL:sanitized_anchor_name:v1.0.0:shurcool_sanitized_anchor_name/vendor/github.com/shurcooL/sanitized_anchor_name \
 		sirupsen:logrus:v1.4.2:sirupsen_logrus/vendor/github.com/sirupsen/logrus \
 		smallstep:assert:82e2b9b3b262:smallstep_assert/vendor/github.com/smallstep/assert \
-		smallstep:certificates:v0.15.1:smallstep_certificates/vendor/github.com/smallstep/certificates \
-		smallstep:certinfo:v1.3.0:smallstep_certinfo/vendor/github.com/smallstep/certinfo \
-		smallstep:crypto:v0.2.0:smallstep_crypto/vendor/go.step.sm/crypto \
+		smallstep:certificates:v0.15.5:smallstep_certificates/vendor/github.com/smallstep/certificates \
+		smallstep:certinfo:v1.4.0:smallstep_certinfo/vendor/github.com/smallstep/certinfo \
+		smallstep:crypto:v0.6.1:smallstep_crypto/vendor/go.step.sm/crypto \
 		smallstep:nosql:v0.3.0:smallstep_nosql/vendor/github.com/smallstep/nosql \
 		smallstep:truststore:v0.9.6:smallstep_truststore/vendor/github.com/smallstep/truststore \
 		smallstep:zcrypto:fbc32cf76bce:smallstep_zcrypto/vendor/github.com/smallstep/zcrypto \
@@ -173,7 +183,7 @@ GH_TUPLE=	\
 		spf13:pflag:v1.0.5:spf13_pflag/vendor/github.com/spf13/pflag \
 		spf13:viper:v1.6.1:spf13_viper/vendor/github.com/spf13/viper \
 		sqs:pbtypes:v1.0.0:sqs_pbtypes/vendor/sourcegraph.com/sqs/pbtypes \
-		square:go-jose:v2.4.0:square_go_jose/vendor/gopkg.in/square/go-jose.v2 \
+		square:go-jose:v2.5.1:square_go_jose/vendor/gopkg.in/square/go-jose.v2 \
 		stretchr:objx:v0.2.0:stretchr_objx/vendor/github.com/stretchr/objx \
 		stretchr:testify:v1.5.1:stretchr_testify/vendor/github.com/stretchr/testify \
 		subosito:gotenv:v1.2.0:subosito_gotenv/vendor/github.com/subosito/gotenv \
@@ -189,10 +199,14 @@ GH_TUPLE=	\
 		urfave:cli:v1.22.2:urfave_cli/vendor/github.com/urfave/cli \
 		uudashr:gocognit:v1.0.1:uudashr_gocognit/vendor/github.com/uudashr/gocognit \
 		weppos:publicsuffix-go:v0.4.0:weppos_publicsuffix_go/vendor/github.com/weppos/publicsuffix-go \
-		xiang90:probing:43a291ad63a2:xiang90_probing/vendor/github.com/xiang90/probing
+		xiang90:probing:43a291ad63a2:xiang90_probing/vendor/github.com/xiang90/probing \
+		sslmate:go-pkcs12:52f69702a001:sslmate_pkcs12/vendor/software.sslmate.com/src/go-pkcs12
 
 GL_TUPLE=	https://gitlab.howett.net:go:plist:591f970eefbbeb04d7b37f334a0c4c3256e32876:go_plist/vendor/howett.net/plist
 
+		# Mirrors for the following packages are not currently known, please look them up and handle these tuples manually:
+		#::52f69702a001:group_name/vendor/software.sslmate.com/src/go-pkcs12 (from software.sslmate.com/src/go-pkcs12@52f69702a001)
+
 SHEBANG_FILES=	autocomplete/bash_autocomplete
 
 GO_TARGET=	./cmd/step
diff --git a/security/step-cli/distinfo b/security/step-cli/distinfo
index 4048d1e77ebc..3b86d784c5b0 100644
--- a/security/step-cli/distinfo
+++ b/security/step-cli/distinfo
@@ -1,6 +1,6 @@
-TIMESTAMP = 1599630329
-SHA256 (smallstep-cli-v0.15.2_GH0.tar.gz) = 936c59243982b9c05b47bdf55738b9583f26c837fc634d04e00038ce32882675
-SIZE (smallstep-cli-v0.15.2_GH0.tar.gz) = 476601
+TIMESTAMP = 1608500128
+SHA256 (smallstep-cli-v0.15.3_GH0.tar.gz) = bb626281a4c4558f421d2c1b97659e51bf149cf6f6deb310f73b6a5dfc0b548b
+SIZE (smallstep-cli-v0.15.3_GH0.tar.gz) = 491529
 SHA256 (AndreasBriese-bbloom-e2d15f34fcf9_GH0.tar.gz) = e88bd873a2251d70b5d2cbf9dceff24fa25ca5652ac1f99165c303e98df494a1
 SIZE (AndreasBriese-bbloom-e2d15f34fcf9_GH0.tar.gz) = 7704
 SHA256 (BurntSushi-toml-v0.3.1_GH0.tar.gz) = 6593da894578ba510a470735ffbdc88ce88033094dc5a8f4d3957ab87e18803f
@@ -29,6 +29,8 @@ SHA256 (bombsimon-wsl-v2.0.0_GH0.tar.gz) = 1beae51b976a358e312986280f5bd73dddbf0
 SIZE (bombsimon-wsl-v2.0.0_GH0.tar.gz) = 24454
 SHA256 (boombuler-barcode-v1.0.0_GH0.tar.gz) = 22b54374186260810959e1ebdc70d80d606f7f73053dbcbf353b001653aae57c
 SIZE (boombuler-barcode-v1.0.0_GH0.tar.gz) = 62930
+SHA256 (census-instrumentation-opencensus-go-v0.22.4_GH0.tar.gz) = f230d965b9ddf15178df13209527c3e15c96cafa1c8a7a1b7eb39d0c424baf37
+SIZE (census-instrumentation-opencensus-go-v0.22.4_GH0.tar.gz) = 170499
 SHA256 (cespare-xxhash-v1.1.0_GH0.tar.gz) = 9418be390574092f0ca989b9ae2f5450270ead8125a635a100850b28c8c9c495
 SIZE (cespare-xxhash-v1.1.0_GH0.tar.gz) = 8200
 SHA256 (cespare-xxhash-v2.1.1_GH0.tar.gz) = 0ee31178d2c5a1249be4e26294a2f428008dc4e1ecbbfbe47f74e41026df1148
@@ -65,8 +67,8 @@ SHA256 (dgrijalva-jwt-go-v3.2.0_GH0.tar.gz) = 197465ef53219f3aeb1a6940b70e16d288
 SIZE (dgrijalva-jwt-go-v3.2.0_GH0.tar.gz) = 36960
 SHA256 (dgryski-go-farm-6a90982ecee2_GH0.tar.gz) = 47bda738d281105cc7b1cb832880f93d220eae40b18095dc322b0b7bf44fdf23
 SIZE (dgryski-go-farm-6a90982ecee2_GH0.tar.gz) = 26800
-SHA256 (dominikh-go-tools-v0.0.1-2019.2.3_GH0.tar.gz) = 3b16c5bec68f71d13be0216b849ce910aa4a1844db6e08dafe90fb1a439974c7
-SIZE (dominikh-go-tools-v0.0.1-2019.2.3_GH0.tar.gz) = 367559
+SHA256 (dominikh-go-tools-v0.0.1-2020.1.4_GH0.tar.gz) = ac3b16836cec7ebe2ab426d84fbfd7946b43ee5ea72f608c2222b13a1e898fc7
+SIZE (dominikh-go-tools-v0.0.1-2020.1.4_GH0.tar.gz) = 406057
 SHA256 (dustin-go-humanize-v1.0.0_GH0.tar.gz) = e4540bd50ac855143b4f2e509313079c50cf5d8774f09cc10dbca5ae9803d8ba
 SIZE (dustin-go-humanize-v1.0.0_GH0.tar.gz) = 17260
 SHA256 (etcd-io-bbolt-v1.3.2_GH0.tar.gz) = 0f4bd88cce84f7b42f6364fc8c77ae7dd7d2f70224d1bb2abf410fc2f552c9a9
@@ -109,30 +111,38 @@ SHA256 (gofrs-flock-5135e617513b_GH0.tar.gz) = 59644ab45696775819ced0299ff5a6aac
 SIZE (gofrs-flock-5135e617513b_GH0.tar.gz) = 7290
 SHA256 (gogo-protobuf-v1.3.1_GH0.tar.gz) = 5184f06decd681fcc82f6583976111faf87189c0c2f8063b34ac2ea9ed997236
 SIZE (gogo-protobuf-v1.3.1_GH0.tar.gz) = 2038188
+SHA256 (golang-appengine-v1.6.6_GH0.tar.gz) = 0c3d1e1c7ba0b97ea3457ddd169aac0667dbbb8c8d81e011632751f75bfbb624
+SIZE (golang-appengine-v1.6.6_GH0.tar.gz) = 332981
 SHA256 (golang-crypto-123391ffb6de_GH0.tar.gz) = 09cadc5885fb57e7c3e3cdfd245a4dffb103adc90eb7f97efbc106905293c538
 SIZE (golang-crypto-123391ffb6de_GH0.tar.gz) = 1732577
 SHA256 (golang-glog-23def4e6c14b_GH0.tar.gz) = 528b6072aa1c5dc69325bd6f057940ba8908703542ec5689b64b98e72c48588b
 SIZE (golang-glog-23def4e6c14b_GH0.tar.gz) = 19660
-SHA256 (golang-groupcache-215e87163ea7_GH0.tar.gz) = 500b097a42fb5c0cd516f1bb56e9c745ba6c1c910b4dc7296aea2b9120ee5e70
-SIZE (golang-groupcache-215e87163ea7_GH0.tar.gz) = 26040
-SHA256 (golang-lint-fdd1cda4f05f_GH0.tar.gz) = d02215cdf7a8c9e1874406b0414f3799e13e235ed4de142f837f68aa69d6fd55
-SIZE (golang-lint-fdd1cda4f05f_GH0.tar.gz) = 31573
-SHA256 (golang-mock-v1.3.1_GH0.tar.gz) = c70de449e2fe826047a87bf25a81517dd3ac0665cdbdfd94ded69b49fe1b0361
-SIZE (golang-mock-v1.3.1_GH0.tar.gz) = 48942
-SHA256 (golang-net-16171245cfb2_GH0.tar.gz) = 941e5a2afb85a131b42f8503fe1a760447a40a0f26539c11ca489f4f24a77e77
-SIZE (golang-net-16171245cfb2_GH0.tar.gz) = 1172446
-SHA256 (golang-protobuf-v1.3.2_GH0.tar.gz) = c9cda622857a17cf0877c5ba76688a931883e505f40744c9495638b6e3da1f65
-SIZE (golang-protobuf-v1.3.2_GH0.tar.gz) = 312285
+SHA256 (golang-groupcache-8c9f03a8e57e_GH0.tar.gz) = b92f918daa48048fd360f14d1a4aed6e70c1176ae6b00b0dc04094bb088e9865
+SIZE (golang-groupcache-8c9f03a8e57e_GH0.tar.gz) = 26047
+SHA256 (golang-lint-738671d3881b_GH0.tar.gz) = c3c30939f0592da3a4b470afd40cdf2924a2749e251b4188f53a918445fb67b4
+SIZE (golang-lint-738671d3881b_GH0.tar.gz) = 31848
+SHA256 (golang-mock-v1.4.4_GH0.tar.gz) = a3e25a2c234f82f6685e143681e223462bd1cc0098375a48dc042e227f2cd677
+SIZE (golang-mock-v1.4.4_GH0.tar.gz) = 55141
+SHA256 (golang-mod-v0.3.0_GH0.tar.gz) = bc97bb300928a00cfe64d97b3240c760668060cda0767e0d0b413e506fc3fa0a
+SIZE (golang-mod-v0.3.0_GH0.tar.gz) = 93941
+SHA256 (golang-net-c89045814202_GH0.tar.gz) = 9e67378e3fb5859927b4a1cc458d87e210859823d14955ba6477c7c6ca4cbcee
+SIZE (golang-net-c89045814202_GH0.tar.gz) = 1178563
+SHA256 (golang-oauth2-bf48bf16ab8d_GH0.tar.gz) = 28b3a51a8bd90beb5952d1080218ea238127edfe9455ad48921989619ff918a8
+SIZE (golang-oauth2-bf48bf16ab8d_GH0.tar.gz) = 47016
+SHA256 (golang-protobuf-v1.4.2_GH0.tar.gz) = a1e486b3ed6f502c43e10cb28bde4e8f020a420ef9958d3fbe61801492c07c87
+SIZE (golang-protobuf-v1.4.2_GH0.tar.gz) = 171778
 SHA256 (golang-snappy-v0.0.1_GH0.tar.gz) = b1d97f47fcb61cb0cdd54bc424eda980c47838effb0ec9e322506514a50fee85
 SIZE (golang-snappy-v0.0.1_GH0.tar.gz) = 62605
-SHA256 (golang-sys-b016eb3dc98e_GH0.tar.gz) = 09d8ddfb8f2c651e956a534d5de1e5401567d12dd0a589abd4ad5e9fa4213b06
-SIZE (golang-sys-b016eb3dc98e_GH0.tar.gz) = 1535262
-SHA256 (golang-text-v0.3.2_GH0.tar.gz) = 0b9309698f5708531c5377ab1e29b423a6d9e20c55a8d386c3b8283428212f22
-SIZE (golang-text-v0.3.2_GH0.tar.gz) = 7168069
+SHA256 (golang-sys-157a740278f4_GH0.tar.gz) = f904c2f9180a6d1e2a4e1e1e4ebcf73b4e293c146999d4473a6364f32680932d
+SIZE (golang-sys-157a740278f4_GH0.tar.gz) = 1059282
+SHA256 (golang-text-v0.3.3_GH0.tar.gz) = 1604233637e3593749fbbb13b5069b08e6feba6d2b55a02fd3148793d5871185
+SIZE (golang-text-v0.3.3_GH0.tar.gz) = 7747332
 SHA256 (golang-time-555d28b269f0_GH0.tar.gz) = 8ce368da65025dad4e4ed302a5835e2b3579f3a111721f9b275803519a5d10b3
 SIZE (golang-time-555d28b269f0_GH0.tar.gz) = 9577
-SHA256 (golang-tools-7be0a674c9fc_GH0.tar.gz) = ce0f3ed9ca433fc7bb8b045ef854c62ec05a71717a4b6f5554dc3cbfae16e9f4
-SIZE (golang-tools-7be0a674c9fc_GH0.tar.gz) = 2316458
+SHA256 (golang-tools-af4cc2cd812e_GH0.tar.gz) = 397e7a83a30d873bbbaa3233f9b94c4acee1cb63f32fdd6d02c59055f9008f21
+SIZE (golang-tools-af4cc2cd812e_GH0.tar.gz) = 2562111
+SHA256 (golang-xerrors-5ec99f83aff1_GH0.tar.gz) = 71975d658357e170fd6a41f92539cde8b39c9cd8bfe5931b6311bc5f5c0da0d7
+SIZE (golang-xerrors-5ec99f83aff1_GH0.tar.gz) = 13664
 SHA256 (golangci-check-cfe4005ccda2_GH0.tar.gz) = c8379c7a28b77302a2a1d88dbdd549037ed34a983851b0eb1e0f186e9895af3d
 SIZE (golangci-check-cfe4005ccda2_GH0.tar.gz) = 16555
 SHA256 (golangci-dupl-3e9179ac440a_GH0.tar.gz) = f56e3ca1b26fbb7a4d84bce6ca2c885043bb8c0fd1ab49ef85d5df02a8afefe2
@@ -167,12 +177,20 @@ SHA256 (google-btree-v1.0.0_GH0.tar.gz) = 9f53e0e1330a4dbe92100d516606791f7196bf
 SIZE (google-btree-v1.0.0_GH0.tar.gz) = 15292
 SHA256 (google-certificate-transparency-go-v1.1.0_GH0.tar.gz) = 78624143842dcdbf9965f6778f8908316b542ad4a04e377d30d9b710fc48f486
 SIZE (google-certificate-transparency-go-v1.1.0_GH0.tar.gz) = 1353276
-SHA256 (google-go-genproto-f3c370f40bfb_GH0.tar.gz) = 754588f44bdfdbe4521ad0950375a0357f2454ff6de43e0e6d6a88b5b5182c55
-SIZE (google-go-genproto-f3c370f40bfb_GH0.tar.gz) = 5857725
+SHA256 (google-go-cmp-v0.5.2_GH0.tar.gz) = 76e0c4238e7fec1490ef2c8b4719a024ade7f4b8778965acb888566535ee9dd5
+SIZE (google-go-cmp-v0.5.2_GH0.tar.gz) = 99769
+SHA256 (google-go-genproto-0bd0a958aa1d_GH0.tar.gz) = 7c6130c31b6e20ef371de46f9119c5ff36f5a89a68598fe76f138e2610468e0e
+SIZE (google-go-genproto-0bd0a958aa1d_GH0.tar.gz) = 12849218
 SHA256 (google-monologue-35abc9683a6c_GH0.tar.gz) = a99e5a2cac29de6fa449b271813ec36b3946623ed486e506b999afb658845f69
 SIZE (google-monologue-35abc9683a6c_GH0.tar.gz) = 48374
-SHA256 (google-uuid-v1.1.1_GH0.tar.gz) = bebd4b0b4ea152a9793615ef23c83f688876d8c284a2092264d20a4bf4ffc423
-SIZE (google-uuid-v1.1.1_GH0.tar.gz) = 13543
+SHA256 (google-uuid-v1.1.2_GH0.tar.gz) = e650558e314307cf33391d0a9ef575b418188206d61cb9751e9f11bceb0874d0
+SIZE (google-uuid-v1.1.2_GH0.tar.gz) = 13871
+SHA256 (googleapis-gax-go-v2.0.5_GH0.tar.gz) = 3089affe6f5e27f7a6d494cb399aa6baf232384f763f548ad5ddfbea0e88e59c
+SIZE (googleapis-gax-go-v2.0.5_GH0.tar.gz) = 15328
+SHA256 (googleapis-google-api-go-client-v0.31.0_GH0.tar.gz) = 2b65d8e9f7c3b036fb945c88bb12da5f465f851b452857420b314c909f5867f9
+SIZE (googleapis-google-api-go-client-v0.31.0_GH0.tar.gz) = 16056013
+SHA256 (googleapis-google-cloud-go-3c2db50b5678_GH0.tar.gz) = 1504fda439df6e15e460484d6cb19a99ddfeb7ceea4a723011120ba988e9106c
+SIZE (googleapis-google-cloud-go-3c2db50b5678_GH0.tar.gz) = 2912178
 SHA256 (gorilla-websocket-v1.4.1_GH0.tar.gz) = 9e5bd337d58a978bf4072767252ddb394b53e2b459697633794abd96e8eb3fa6
 SIZE (gorilla-websocket-v1.4.1_GH0.tar.gz) = 54218
 SHA256 (gostaticanalysis-analysisutil-v0.0.3_GH0.tar.gz) = 991512e6bda62847eaf77aec69deca4efc1bf44775b8cb2c4571133f8f8fd750
@@ -183,8 +201,8 @@ SHA256 (grpc-ecosystem-go-grpc-prometheus-v1.2.0_GH0.tar.gz) = eba66530952a126ab
 SIZE (grpc-ecosystem-go-grpc-prometheus-v1.2.0_GH0.tar.gz) = 24760
 SHA256 (grpc-ecosystem-grpc-gateway-v1.12.1_GH0.tar.gz) = 8a3ce7fa981dd21c0693fb2922457053a7c2e5a5687e22d9f19fe7c35520c35e
 SIZE (grpc-ecosystem-grpc-gateway-v1.12.1_GH0.tar.gz) = 438424
-SHA256 (grpc-grpc-go-v1.26.0_GH0.tar.gz) = a594cbd8f7d545d181c92b27aafd5d4824459e3a729a8bd67a0c8b99c411f05a
-SIZE (grpc-grpc-go-v1.26.0_GH0.tar.gz) = 765416
+SHA256 (grpc-grpc-go-v1.29.1_GH0.tar.gz) = b626c128878d1d7aaf79c6f4f73cba0eadd26405a30fc00b71cc9a12e56fef38
+SIZE (grpc-grpc-go-v1.29.1_GH0.tar.gz) = 892944
 SHA256 (hashicorp-hcl-v1.0.0_GH0.tar.gz) = 50632428210503070fd2fde748c88b7414bf84a6a0eadebf9d8e596a033bead2
 SIZE (hashicorp-hcl-v1.0.0_GH0.tar.gz) = 70658
 SHA256 (huandu-xstrings-v1.3.1_GH0.tar.gz) = d399f03735391073441145bac6b8d06b36f3b59e005db77ebafde130ddf215bf
@@ -267,6 +285,8 @@ SHA256 (prometheus-common-v0.7.0_GH0.tar.gz) = 71b1dce2afd959f9fcab3b5e097dfa9d3
 SIZE (prometheus-common-v0.7.0_GH0.tar.gz) = 100520
 SHA256 (prometheus-procfs-v0.0.8_GH0.tar.gz) = 1c26c8b86a9b1902684f860f94d0bf4b94abcc570e456bf9a74d1c9d83643872
 SIZE (prometheus-procfs-v0.0.8_GH0.tar.gz) = 126299
+SHA256 (protocolbuffers-protobuf-go-v1.25.0_GH0.tar.gz) = c1c04d6e36c0d0fb6f3374197f9025d7e6df13f38a974098be020617c00fbaf2
+SIZE (protocolbuffers-protobuf-go-v1.25.0_GH0.tar.gz) = 1258804
 SHA256 (rs-xid-v1.2.1_GH0.tar.gz) = bb207227d5ae99bda71d38ae11e29b822c9b572223781bc282ad2f8e69002f2c
 SIZE (rs-xid-v1.2.1_GH0.tar.gz) = 9553
 SHA256 (russross-blackfriday-v2.0.1_GH0.tar.gz) = 5a0f38a36b6f3b2d59b72d713451a895a4d3a4406b3533882483782e37797cff
@@ -281,12 +301,12 @@ SHA256 (sirupsen-logrus-v1.4.2_GH0.tar.gz) = 67f2ddf467b7e63d2d2529d227946a331e2
 SIZE (sirupsen-logrus-v1.4.2_GH0.tar.gz) = 41373
 SHA256 (smallstep-assert-82e2b9b3b262_GH0.tar.gz) = 6c19763fe97ee07940c12a5ed7693865146e90ab6862deea4f5e907cac7d4880
 SIZE (smallstep-assert-82e2b9b3b262_GH0.tar.gz) = 4069
-SHA256 (smallstep-certificates-v0.15.1_GH0.tar.gz) = 354b36c1c34b90f2739e943e4f0a6dc102f2d05be146ebb2e6bbb0794478167f
-SIZE (smallstep-certificates-v0.15.1_GH0.tar.gz) = 17584657
-SHA256 (smallstep-certinfo-v1.3.0_GH0.tar.gz) = 58f20874b73f1f363f14e5fbcbd0fc4e030045cbc5299faee8f8967ea699a9cf
-SIZE (smallstep-certinfo-v1.3.0_GH0.tar.gz) = 58609
-SHA256 (smallstep-crypto-v0.2.0_GH0.tar.gz) = 8fab9f50ccc6cd88147f718afd477f7957e61a9d66ec0eb9380a7b5521e6a7fa
-SIZE (smallstep-crypto-v0.2.0_GH0.tar.gz) = 100980
+SHA256 (smallstep-certificates-v0.15.5_GH0.tar.gz) = daf2360f6e6f89febacc0a29453302a827e10b2638029818ac18dc66da479d6e
+SIZE (smallstep-certificates-v0.15.5_GH0.tar.gz) = 17622399
+SHA256 (smallstep-certinfo-v1.4.0_GH0.tar.gz) = d56783c6933ad7fd39749c2184ab249468461e03ce0fa26fdf8d33cf02217e5c
+SIZE (smallstep-certinfo-v1.4.0_GH0.tar.gz) = 58640
+SHA256 (smallstep-crypto-v0.6.1_GH0.tar.gz) = 5803ba189d26da1fb2b071ae344b4b97aee6ce5228cf253ad818b59305934bae
+SIZE (smallstep-crypto-v0.6.1_GH0.tar.gz) = 133334
 SHA256 (smallstep-nosql-v0.3.0_GH0.tar.gz) = 6020dc16800b55f6fd83be95e0e5c9d0d1ed4aee530a3510e06cd8ce7aa4b470
 SIZE (smallstep-nosql-v0.3.0_GH0.tar.gz) = 21961
 SHA256 (smallstep-truststore-v0.9.6_GH0.tar.gz) = 31180e68b8731a9c091e31059632b41eabbee9d78ae91cd1fe1ee97a25ddf358
@@ -313,8 +333,8 @@ SHA256 (spf13-viper-v1.6.1_GH0.tar.gz) = 6c31cd5d8fffd6736589737e4e12f770fde4647
 SIZE (spf13-viper-v1.6.1_GH0.tar.gz) = 51749
 SHA256 (sqs-pbtypes-v1.0.0_GH0.tar.gz) = 64c42ffbc8ce815319a53f4e849988389e20c4633a6eadbe0d6ed225a949baf3
 SIZE (sqs-pbtypes-v1.0.0_GH0.tar.gz) = 9153
-SHA256 (square-go-jose-v2.4.0_GH0.tar.gz) = 3824f79c5f851784032a7800d72f32229ce6617612fab28642c18b5150b52493
-SIZE (square-go-jose-v2.4.0_GH0.tar.gz) = 303589
+SHA256 (square-go-jose-v2.5.1_GH0.tar.gz) = 74c65592183c542b254eb2933f7a99ee869abdf9e7ac02aad4d9f0dce980ace8
+SIZE (square-go-jose-v2.5.1_GH0.tar.gz) = 309860
 SHA256 (stretchr-objx-v0.2.0_GH0.tar.gz) = be7dce02798cdc24cee6bd2d95bfcf6976b1521ef735349bf5e7dfe6d147746f
 SIZE (stretchr-objx-v0.2.0_GH0.tar.gz) = 79981
 SHA256 (stretchr-testify-v1.5.1_GH0.tar.gz) = 6d65f92aeff186bb261dbdb885d9f3ba09db3d79eb5f212fcace1fd4a2858263
@@ -347,3 +367,5 @@ SHA256 (weppos-publicsuffix-go-v0.4.0_GH0.tar.gz) = 52ea269f2b3e2cd3240dc1d6e49a
 SIZE (weppos-publicsuffix-go-v0.4.0_GH0.tar.gz) = 54614
 SHA256 (xiang90-probing-43a291ad63a2_GH0.tar.gz) = 576dd87a3b8729fd6f2422664897e285d43254138831aaedfac797813964d255
 SIZE (xiang90-probing-43a291ad63a2_GH0.tar.gz) = 3468
+SHA256 (sslmate-go-pkcs12-52f69702a001_GH0.tar.gz) = 80ebd46e3edb65a73e7176569cfdc3992861332d8b966eae032da07348415717
+SIZE (sslmate-go-pkcs12-52f69702a001_GH0.tar.gz) = 26621
