diff --git a/sysutils/R-cran-processx/Makefile b/sysutils/R-cran-processx/Makefile
index e3e278d785a9..259dd8713af0 100644
--- a/sysutils/R-cran-processx/Makefile
+++ b/sysutils/R-cran-processx/Makefile
@@ -1,5 +1,5 @@
 PORTNAME=	processx
-DISTVERSION=	3.4.5
+DISTVERSION=	3.5.1
 CATEGORIES=	sysutils
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
@@ -11,8 +11,9 @@ LICENSE_FILE=	${WRKSRC}/LICENSE
 
 RUN_DEPENDS=	R-cran-R6>0:devel/R-cran-R6 \
 		R-cran-ps>=1.2.0:sysutils/R-cran-ps
-TEST_DEPENDS=	R-cran-covr>0:devel/R-cran-covr \
-		R-cran-crayon>0:devel/R-cran-crayon \
+TEST_DEPENDS=	R-cran-callr>=3.2.0:devel/R-cran-callr \
+		R-cran-cli>0:devel/R-cran-cli \
+		R-cran-covr>0:devel/R-cran-covr \
 		R-cran-testthat>0:devel/R-cran-testthat \
 		R-cran-withr>0:devel/R-cran-withr \
 		R-cran-curl>0:ftp/R-cran-curl
diff --git a/sysutils/R-cran-processx/distinfo b/sysutils/R-cran-processx/distinfo
index 6b0344dbae4a..2ac2c655f2e4 100644
--- a/sysutils/R-cran-processx/distinfo
+++ b/sysutils/R-cran-processx/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1609275213
-SHA256 (processx_3.4.5.tar.gz) = e368103aa6a6894bfa8e78b12a25598464bcd2c19a8b6334f24ee397db13bb14
-SIZE (processx_3.4.5.tar.gz) = 135121
+TIMESTAMP = 1618642606
+SHA256 (processx_3.5.1.tar.gz) = d35c13f131692573939949f21be03988c12bfc18d4b5093fa43d10f0ce53fffc
+SIZE (processx_3.5.1.tar.gz) = 138959
