diff --git a/sysutils/R-cran-processx/Makefile b/sysutils/R-cran-processx/Makefile
index 259dd8713af0..606e0264a7c5 100644
--- a/sysutils/R-cran-processx/Makefile
+++ b/sysutils/R-cran-processx/Makefile
@@ -1,5 +1,5 @@
 PORTNAME=	processx
-DISTVERSION=	3.5.1
+DISTVERSION=	3.5.2
 CATEGORIES=	sysutils
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
diff --git a/sysutils/R-cran-processx/distinfo b/sysutils/R-cran-processx/distinfo
index 2ac2c655f2e4..051ca11d1afa 100644
--- a/sysutils/R-cran-processx/distinfo
+++ b/sysutils/R-cran-processx/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1618642606
-SHA256 (processx_3.5.1.tar.gz) = d35c13f131692573939949f21be03988c12bfc18d4b5093fa43d10f0ce53fffc
-SIZE (processx_3.5.1.tar.gz) = 138959
+TIMESTAMP = 1620101366
+SHA256 (processx_3.5.2.tar.gz) = ed6f2d1047461c6061e6ed58fb6de65a289b56009867892abad76c6bba46fc2b
+SIZE (processx_3.5.2.tar.gz) = 141424
