diff --git a/sysutils/R-cran-ps/Makefile b/sysutils/R-cran-ps/Makefile
index 15ba305e2eb3..c11c2104097f 100644
--- a/sysutils/R-cran-ps/Makefile
+++ b/sysutils/R-cran-ps/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
 PORTNAME=	ps
-DISTVERSION=	1.5.0
+DISTVERSION=	1.6.0
 CATEGORIES=	sysutils
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
@@ -12,11 +12,13 @@ LICENSE=	MIT
 LICENSE_FILE=	${WRKSRC}/LICENSE
 
 TEST_DEPENDS=	R-cran-R6>0:devel/R-cran-R6 \
+		R-cran-callr>0:devel/R-cran-callr \
 		R-cran-covr>0:devel/R-cran-covr \
 		R-cran-rlang>0:devel/R-cran-rlang \
 		R-cran-testthat>0:devel/R-cran-testthat \
 		R-cran-tibble>0:devel/R-cran-tibble \
-		R-cran-curl>0:ftp/R-cran-curl
+		R-cran-curl>0:ftp/R-cran-curl \
+		R-cran-processx>=3.1.0:sysutils/R-cran-processx
 
 USES=		cran:auto-plist,compiles
 
diff --git a/sysutils/R-cran-ps/distinfo b/sysutils/R-cran-ps/distinfo
index d6bf7d195978..ea4a7b358c62 100644
--- a/sysutils/R-cran-ps/distinfo
+++ b/sysutils/R-cran-ps/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1608946306
-SHA256 (ps_1.5.0.tar.gz) = 7461a196f55557feda569a9791ad851c884f9a2dd71671655ed17cb048fafe96
-SIZE (ps_1.5.0.tar.gz) = 115131
+TIMESTAMP = 1615084795
+SHA256 (ps_1.6.0.tar.gz) = 89ad7ddc5e0818bccacfd0673ddf2da0892ac2a3b4d3a821e40884ab1e96bf31
+SIZE (ps_1.6.0.tar.gz) = 119670
