diff --git a/textproc/R-cran-xmlparsedata/Makefile b/textproc/R-cran-xmlparsedata/Makefile
index cf2f59e9fb55..16788b43756e 100644
--- a/textproc/R-cran-xmlparsedata/Makefile
+++ b/textproc/R-cran-xmlparsedata/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$
 
 PORTNAME=	xmlparsedata
-DISTVERSION=	1.0.4
+DISTVERSION=	1.0.5
 CATEGORIES=	textproc
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
diff --git a/textproc/R-cran-xmlparsedata/distinfo b/textproc/R-cran-xmlparsedata/distinfo
index 9e7820123ace..7ee3a00af59e 100644
--- a/textproc/R-cran-xmlparsedata/distinfo
+++ b/textproc/R-cran-xmlparsedata/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1609275995
-SHA256 (xmlparsedata_1.0.4.tar.gz) = 387b13c25bea9ddc0a39b817c17c199b86ab9acafa328daae2233a9ca577fb9c
-SIZE (xmlparsedata_1.0.4.tar.gz) = 8399
+TIMESTAMP = 1615085011
+SHA256 (xmlparsedata_1.0.5.tar.gz) = 766034ab5e9728609bd240c9954d23ca0cdb881a98a31b9d3e1c8767c7b7cbb0
+SIZE (xmlparsedata_1.0.5.tar.gz) = 8993
