diff --git a/textproc/R-cran-rex/Makefile b/textproc/R-cran-rex/Makefile
index 4575664d4a75..d0a0abb52e44 100644
--- a/textproc/R-cran-rex/Makefile
+++ b/textproc/R-cran-rex/Makefile
@@ -1,12 +1,11 @@
 # $FreeBSD$
 
 PORTNAME=	rex
-DISTVERSION=	1.1.2
-PORTREVISION=	3
+DISTVERSION=	1.2.0
 CATEGORIES=	textproc
 DISTNAME=	${PORTNAME}_${DISTVERSION}
 
-MAINTAINER=	ports@FreeBSD.org
+MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Friendly Regular Expressions
 
 LICENSE=	MIT
diff --git a/textproc/R-cran-rex/distinfo b/textproc/R-cran-rex/distinfo
index a7118ca1fb55..724694ae5f09 100644
--- a/textproc/R-cran-rex/distinfo
+++ b/textproc/R-cran-rex/distinfo
@@ -1,3 +1,3 @@
-TIMESTAMP = 1540317825
-SHA256 (rex_1.1.2.tar.gz) = bd3c74ceaf335336f5dd04314d0a791f6311e421a2158f321f5aab275f539a2a
-SIZE (rex_1.1.2.tar.gz) = 89430
+TIMESTAMP = 1606834578
+SHA256 (rex_1.2.0.tar.gz) = 06b491f1469078862e40543fd74e1d38b2e0fb61fdf01c8083add4b11ac2eb54
+SIZE (rex_1.2.0.tar.gz) = 91925
