diff --git a/www/cpr/Makefile b/www/cpr/Makefile
index 017cac323126..9676c7fa258e 100644
--- a/www/cpr/Makefile
+++ b/www/cpr/Makefile
@@ -1,12 +1,7 @@
 PORTNAME=	cpr
-DISTVERSION=	1.3.0-57
-PORTREVISION=	1
-DISTVERSIONSUFFIX=	-g3d14e61
+PORTVERSION=	1.6.2
 CATEGORIES=	www
 
-PATCH_SITES=	https://github.com/${GH_ACCOUNT}/${GH_PROJECT}/commit/
-PATCHFILES=	8eb16e267843d18834bf42d9ae4ccfb58f34bdb5.patch:-p1
-
 MAINTAINER=	ygy@FreeBSD.org
 COMMENT=	Curl for people, a simple wrapper around libcurl
 
@@ -15,15 +10,12 @@ LICENSE_FILE=	${WRKSRC}/LICENSE
 
 LIB_DEPENDS=	libcurl.so:ftp/curl
 
-USES=		compiler:c++11-lang cmake
+USES=		cmake compiler:c++11-lang ssl
+
 USE_GITHUB=	yes
 GH_ACCOUNT=	whoshuu
-GH_TUPLE=	whoshuu:mongoose:df9f7a7:mongoose/opt/mongoose
-USE_LDCONFIG=	yes
-
-CMAKE_OFF=	BUILD_CPR_TESTS
-CMAKE_ON=	BUILD_SHARED_LIBS USE_SYSTEM_CURL
 
-INSTALL_TARGET=	install
+CMAKE_OFF=	CPR_BUILD_TESTS CPR_BUILD_TESTS_SSL CPR_CURL_NOSIGNAL CPR_FORCE_DARWINSSL_BACKEND CPR_FORCE_WINSSL_BACKEND CPR_GENERATE_COVERAGE CPR_USE_SYSTEM_GTEST
+CMAKE_ON=	BUILD_SHARED_LIBS CPR_ENABLE_SSL CPR_FORCE_OPENSSL_BACKEND CPR_FORCE_USE_SYSTEM_CURL
 
 .include <bsd.port.mk>
diff --git a/www/cpr/distinfo b/www/cpr/distinfo
index fb4cb7b9adaf..4002e0146775 100644
--- a/www/cpr/distinfo
+++ b/www/cpr/distinfo
@@ -1,7 +1,3 @@
-TIMESTAMP = 1533009130
-SHA256 (whoshuu-cpr-1.3.0-57-g3d14e61_GH0.tar.gz) = 14c91766e2f2e2a594212ebcb71b41851402865800f4626559a618a3fd93a2e0
-SIZE (whoshuu-cpr-1.3.0-57-g3d14e61_GH0.tar.gz) = 34678
-SHA256 (whoshuu-mongoose-df9f7a7_GH0.tar.gz) = 3bebbb71bf30a9524dba2a434087965e237c86018906dc2af4eaee8bc273b55f
-SIZE (whoshuu-mongoose-df9f7a7_GH0.tar.gz) = 114812
-SHA256 (8eb16e267843d18834bf42d9ae4ccfb58f34bdb5.patch) = 240d8eee48d5602c1b6dd69f69e33335da642b618a18002514326ae075e1aa47
-SIZE (8eb16e267843d18834bf42d9ae4ccfb58f34bdb5.patch) = 1715
+TIMESTAMP = 1619910295
+SHA256 (whoshuu-cpr-1.6.2_GH0.tar.gz) = c45f9c55797380c6ba44060f0c73713fbd7989eeb1147aedb8723aa14f3afaa3
+SIZE (whoshuu-cpr-1.6.2_GH0.tar.gz) = 68856
diff --git a/www/cpr/files/patch-cpr-error.cpp b/www/cpr/files/patch-cpr-error.cpp
deleted file mode 100644
index 78ac1b4b84a2..000000000000
--- a/www/cpr/files/patch-cpr-error.cpp
+++ /dev/null
@@ -1,27 +0,0 @@
---- cpr/error.cpp.orig	2017-11-01 21:56:02 UTC
-+++ cpr/error.cpp
-@@ -22,8 +22,10 @@ ErrorCode Error::getErrorCodeForCurlErro
-             return ErrorCode::OPERATION_TIMEDOUT;
-         case CURLE_SSL_CONNECT_ERROR:
-             return ErrorCode::SSL_CONNECT_ERROR;
-+#if LIBCURL_VERSION_NUM < 0x073e00
-         case CURLE_PEER_FAILED_VERIFICATION:
-             return ErrorCode::SSL_REMOTE_CERTIFICATE_ERROR;
-+#endif
-         case CURLE_GOT_NOTHING:
-             return ErrorCode::EMPTY_RESPONSE;
-         case CURLE_SSL_ENGINE_NOTFOUND:
-@@ -38,8 +40,13 @@ ErrorCode Error::getErrorCodeForCurlErro
-             return ErrorCode::SSL_LOCAL_CERTIFICATE_ERROR;
-         case CURLE_SSL_CIPHER:
-             return ErrorCode::GENERIC_SSL_ERROR;
-+#if LIBCURL_VERSION_NUM >= 0x073e00
-+        case CURLE_PEER_FAILED_VERIFICATION:
-+            return ErrorCode::SSL_REMOTE_CERTIFICATE_ERROR;
-+#else
-         case CURLE_SSL_CACERT:
-             return ErrorCode::SSL_CACERT_ERROR;
-+#endif
-         case CURLE_USE_SSL_FAILED:
-             return ErrorCode::GENERIC_SSL_ERROR;
-         case CURLE_SSL_ENGINE_INITFAILED:
diff --git a/www/cpr/files/patch-cpr_session.cpp b/www/cpr/files/patch-cpr_session.cpp
deleted file mode 100644
index 67da3040fb32..000000000000
--- a/www/cpr/files/patch-cpr_session.cpp
+++ /dev/null
@@ -1,11 +0,0 @@
---- cpr/session.cpp.orig	2018-07-18 20:02:06 UTC
-+++ cpr/session.cpp
-@@ -350,7 +350,7 @@ Response Session::Impl::Patch() {
- Response Session::Impl::Post() {
-     auto curl = curl_->handle;
-     if (curl) {
--        curl_easy_setopt(curl, CURLOPT_HTTPGET, 0L);
-+        curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "POST");
-         curl_easy_setopt(curl, CURLOPT_NOBODY, 0L);
-     }
- 
diff --git a/www/cpr/pkg-plist b/www/cpr/pkg-plist
index 22eea02bf68c..350587b31e97 100644
--- a/www/cpr/pkg-plist
+++ b/www/cpr/pkg-plist
@@ -1,23 +1,33 @@
 include/cpr/api.h
 include/cpr/auth.h
+include/cpr/bearer.h
 include/cpr/body.h
+include/cpr/callback.h
+include/cpr/connect_timeout.h
 include/cpr/cookies.h
 include/cpr/cpr.h
 include/cpr/cprtypes.h
+include/cpr/curl_container.h
 include/cpr/curlholder.h
-include/cpr/defines.h
 include/cpr/digest.h
 include/cpr/error.h
+include/cpr/limit_rate.h
 include/cpr/low_speed.h
 include/cpr/max_redirects.h
 include/cpr/multipart.h
+include/cpr/ntlm.h
 include/cpr/parameters.h
 include/cpr/payload.h
 include/cpr/proxies.h
 include/cpr/response.h
 include/cpr/session.h
 include/cpr/ssl_options.h
+include/cpr/status_codes.h
 include/cpr/timeout.h
+include/cpr/unix_socket.h
+include/cpr/user_agent.h
 include/cpr/util.h
-lib/cmake/cpr/cpr-config.cmake
+include/cpr/verbose.h
 lib/libcpr.so
+lib/libcpr.so.1
+lib/libcpr.so.1.6
